force-app/main/default/triggers/OpportunityTrigger.trigger

```diff
@@ -0,0 +1,48 @@
+trigger OpportunityTrigger on Opportunity (before update, before delete) {
+  if (trigger.isBefore && trigger.isUpdate) {
+    for (opportunity opp : Trigger.new) {
+      if (opp.Amount < 5000) {
+          opp.addError('Opportunity amount must be greater than 5000');   
+      }
+    }
+  }
+         if (Trigger.isBefore && Trigger.isDelete) {
+          Set <Id> accountIds = new Set<Id>();
+          for (Opportunity opp : Trigger.old) {
+             accountIds.add(opp.AccountId);
+          }
+          Map <Id, Account> accountsWithBanking = new Map<Id, Account>();
+          List<Account> accountsForOpps = [SELECT Id, Industry FROM Account WHERE Id IN :accountIds];
```

Comment:
You can avoid most of this looping by querying into a Map of Opportunities and using the relationship to Account like Account.Industry = 'Banking' to pull the records that are needed. In before context the records should still be "there" to query.